### PR TITLE
Add Luware Nimbus Connector Artifacts

### PR DESCRIPTION
**Pull request by bleul**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/Luware Nimbus/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,2326 @@
...
+          "description": "Defines how long Nimbus tries to reach the Destination before the attempt will be stopped.",
+          "x-ms-summary": "Attempt Timeout in Seconds",
+          "minimum": 30,
+          "maximum": 86400,
+          "default": 3600
+        },
+        "maxQueueTimeInSeconds": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Defines how long Nimbus tries to solve the task until the tasks is considered �Limit Reached�.",
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiDefinition.swagger.json]: Could you please correct the grammar mistake "tasks is"

File: certified-connectors/Luware Nimbus/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,2326 @@
...
+            "$ref": "#/definitions/Address"
+          },
+          "description": "List of addresses."
+        },
+        "customFields": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomField"
+          },
+          "description": "List of custom fields of the ontact."
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiDefinition.swagger.json]: Could you please correct the word "ontact"

File: certified-connectors/Luware Nimbus/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,2326 @@
...
+              "items": {
+                "$ref": "#/definitions/ContactReadDto"
+              }
+            }
+          },
+          "401": {
+            "description": "Unauthorized"
+          },
+          "403": {
+            "description": "Forbidden. No access too Addressbook."
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiDefinition.swagger.json]: Can you please verify is this "too" or "to"

File: certified-connectors/Luware Nimbus/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,2326 @@
...
+            "x-ms-url-encoding": "single",
+            "description": "The Nimbus specific ID of the task.",
+            "x-ms-summary": "Task ID",
+            "type": "string",
+            "required": true
+          },
+          {
+            "name": "TaskInformation",
+            "in": "body",
+            "description": "Contains all fields which can be updated on a Task in Proress.",
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiDefinition.swagger.json]: Can you please correct this word "Proress"

File: certified-connectors/Luware Nimbus/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "aad",
+          "clientId": "<< Please add your client id >>",
+          "scopes": [ "User.Read" ],
+          "redirectMode": "Global",
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiProperties.json]: Can you please change this to "redirectMode": "GlobalPerConnector", please refer this link https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#verify-connection-parameters

**Review by vmanoharas — COMMENTED** [review]
Hello @bleul, 

Could you please resolve review comments?

File: certified-connectors/Luware Nimbus/apiProperties.json (hunk 5 above)
Comment (bleul) [reply certified-connectors/Luware Nimbus/apiProperties.json]: Thanks for the corrections. I have updated the PR with all suggested changes.

**Comment by bleul** [comment]
> Hello @bleul,
> 
> Could you please resolve review comments?

Hello @vmanoharas 

Thanks for the corrections. I have updated the PR with all suggested changes.

File: certified-connectors/Luware Nimbus/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+              "value": "https://login.windows.net"
+            },
+            "tenantId": {
+              "value": "common"
+            },
+            "resourceUri": {
+              "value": "<< Please add your client id >>"
+            }
+          },
+          "uiDefinition": {
```
Comment (vmanoharas) [inline certified-connectors/Luware Nimbus/apiProperties.json]: Could you please fix the apiProperties.json issue by referring the schema from this link. https://github.com/microsoft/PowerPlatformConnectors/blob/dev/schemas/paconn-apiProperties.schema.json

**Review by vmanoharas — COMMENTED** [review]
Hello @bleul, 

Could you please refer the link shared and fix apiProperties.json file issues.

File: certified-connectors/Luware Nimbus/apiProperties.json (hunk 6 above)
Comment (bleul) [reply certified-connectors/Luware Nimbus/apiProperties.json]: Hi @vmanoharas 

I have two questions about this. 

1. @{level=Error; message=Value "GlobalPerConnector" is not defined in enum.; valuePath=line: 10 position: 46 path: '[0].properties.connectionParameters.token.oAuthSettings.redirectMode'; value=GlobalPerConnector}
We changed this from "Global" to "GlobalPerConnector", but according to the schema, only "Global" and "Direct" are allowed. The change was made based on your comment on Friday: https://github.com/microsoft/PowerPlatformConnectors/pull/3097#discussion_r1397251539 

2. @{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=The tenant ID of for the Azure Active Directory application}
According to the schema,  "The tenant ID of for the Azure Active Directory application" is the only allowed value here.

Can you please advice on what are the correct values in these two cases, so the file doesn't fail the validation?

Comment (vmanoharas) [reply certified-connectors/Luware Nimbus/apiProperties.json]: Hello @bleul, 

Please accept my apology, if my comment is confusing.  For # 1 please ignore GlobalPerConnector error, this need to be updated from our end. 

For # 2 - could you please remove description and displayname?

image

Comment (bleul) [reply certified-connectors/Luware Nimbus/apiProperties.json]: Hi @vmanoharas 

apiProperties.json was updated accordingly.

Comment (vmanoharas) [reply certified-connectors/Luware Nimbus/apiProperties.json]: Thank you very much @bleul

**Review by vmanoharas — APPROVED** [review]
Hello @bleul 
I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.

**Pull request merged**